src/main/java/randoop/generation/constanttfidf/ConstantMiningStatistics.java

```diff
@@ -0,0 +1,251 @@
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    if (stats == null) {
+      Log.logPrintf("Scope %s is not a key in scopeStatisticsMap%n", scope);
+      return Collections.emptySet();
+    }
+
+    return stats.getSequenceSet();
+  }
+
+  /**
+   * Get the map from every constant to the number of times it is used under all scopes.
+   *
+   * @return the map from every constant to the number of times it is used under all scopes
+   */
+  public Map<Object, Map<Sequence, Integer>> getNumUses() {
+    Map<Object, Map<Sequence, Integer>> res = new HashMap<>();
+    scopeStatisticsMap.forEach((key, value) -> res.put(key, value.getNumUses()));
+    return res;
+  }
+
+  /**
+   * Get the map from every constant to the number of times it is used under the given scope.
+   *
+   * @param scope a type, a package, or the "all" scope
+   * @return the map from every constant to the number of times it is used under the given scop
+   */
+  public Map<Sequence, Integer> getNumUses(Object scope) {
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    if (stats == null) {
+      Log.logPrintf("Scope %s is not a key in scopeStatisticsMap%n", scope);
+      return Collections.emptyMap();
+    }
+    return stats.getNumUses();
+  }
+
+  /**
+   * Get the map from every constant to the number of classes in all scopes that contains it.
+   *
+   * @return the map from every constant to the number of classes in all scopes that contains it
+   */
+  public Map<Object, Map<Sequence, Integer>> getNumClassesWith() {
+    Map<Object, Map<Sequence, Integer>> res = new HashMap<>();
+    scopeStatisticsMap.forEach((key, value) -> res.put(key, value.getNumClassesWith()));
+    return res;
+  }
+
+  /**
+   * Get the map from every constant to the number of classes in the given scope that contains it.
+   *
+   * @param scope a scope
+   * @return the map from every constant to the number of classes in the given scope that contains
+   *     it
+   */
+  public Map<Sequence, Integer> getNumClassesWith(Object scope) {
+    if (GenInputsAbstract.literals_level == ClassLiteralsMode.CLASS) {
+      throw new RandoopBug("Should not get numClassesWith in CLASS level");
+    }
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    if (stats == null) {
+      Log.logPrintf("Scope %s is not a key in scopeStatisticsMap%n", scope);
+      return Collections.emptyMap();
+    }
+    return stats.getNumClassesWith();
+  }
+
+  /**
+   * Get the number of classes in the given scope.
+   *
+   * @param scope a scope
+   * @return the number of classes in the given scope
+   */
+  public Integer getTotalClassesInScope(@Nullable Object scope) {
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    // The default value is null to avoid when scope is java.lang or other standard libraries
+    if (stats == null) {
+      Log.logPrintf("Scope %s is not a key in scopeStatisticsMap%n", scope);
```

Comment:
Should this throw an exception instead?  My real question is about the contract for this method:  Should `scope` be a scope that exists in `scopeStatisticsMap`?  If so, the return type is non-null `Integer`.  If not, the return type should be `@Nullable Integer`.  The question is whether returning null is normal behavior that clients need to check for, or whether clients are responsible for calling this method in a way that guarantees that `null` is never returned.